### PR TITLE
Refactor client gen

### LINKED ISSUE #315: Refactor client-gen
### File Location(s)

client-gen/

### Proposal

split it into manageable chunks and use https://github.com/udoprog/genco or similar

### PR DESCRIPTION
**Pull request by holykol**
* switched to genco (thanks for suggestion @vobradovich)
* move generators to different files for easier navigation
* snapshots are unchanged apart from whitespace changes

Resolves #315

### CODE REVIEW

File: client-gen/Cargo.toml [hunk 1]
```diff
@@ -10,6 +10,7 @@ anyhow.workspace = true
 convert-case.workspace = true
 parity-scale-codec.workspace = true
 sails-idl-parser.workspace = true
+genco = "0.17.9"
```
Comment (vobradovich) [inline client-gen/Cargo.toml:13]: maybe a root cargo.toml would be better?

File: Cargo.toml [hunk 2]
```diff
@@ -36,6 +36,7 @@ convert-case = { package = "convert_case", version = "0.6" }
 fluent-asserter = "0.1"
 futures = { version = "0.3", default-features = false }
 gear-core-errors = "1.4.1"
+genco = "0.17.9"
```
Comment (DennisInSky) [inline Cargo.toml:39]: Minor version should be enough

File: client-gen/Cargo.toml (hunk 1 above)
Comment (holykol) [reply client-gen/Cargo.toml]: thanks, fixed

**Review by DennisInSky — APPROVED** [review]

File: client-gen/src/root_generator.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    ctor_generators::*, io_generators::IoModuleGenerator, service_generators::*, type_generators::*,
+};
+use genco::prelude::*;
+use rust::Tokens;
+use sails_idl_parser::{ast::visitor::Visitor, ast::*};
+
+pub(crate) struct RootGenerator<'a> {
+    tokens: Tokens,
+    traits_tokens: Tokens,
+    default_service_name: &'a str,
```
[the remark this answers is not in the document]
Comment (holykol) [reply client-gen/src/root_generator.rs]: Not quite. default_service name is the name derived from idl file name. I've changed it to anonymous_service_name to (hopefully) avoid confusion

**Pull request merged**